test/server/doc.test.js

```diff
@@ -0,0 +1,481 @@
+            expect(res.body.error).to.eql('Content required');
+            done();
+          });
+      });
+    });
+    describe('READ', () => {
+      it('should GET ONE Documents from api/documents', (done) => {
+        request
+          .get(`/api/documents/${documentId}`)
+          .set('x-access-token', token)
+          .expect('Content-Type', /json/)
+          .end((err, res) => {
+            expect(res.status).to.exist;
+            expect(res.body).to.exist;
+            expect(res.status).to.equal(200);
+            expect(res.body).to.be.a('object');
+            expect(res.body).to.include.keys('document');
+            expect(res.body).to.have.property('document');
+            expect(res.body.status).to.be.true;
+            expect(res.body.document).to.include.keys('_id', 'ownerId', 'content', 'title', 'modifiedAt', 'createdAt');
+            done();
+          });
+      });
+      it('should GET ONE Documents from with a non existent id_ api/documents', (done) => {
+        request
+          .get('/api/documents/57daf1ae48324f06fb4cafd6')
+          .set('x-access-token', token)
+          .expect('Content-Type', /json/)
+          .end((err, res) => {
+            expect(res.status).to.exist;
+            expect(res.body).to.exist;
+            expect(res.status).to.equal(404);
+            expect(res.body).to.be.a('object');
+            expect(res.body).to.have.property('document');
+            expect(res.body.status).to.be.true;
+            expect(res.body.document).to.be.null;
+            done();
+          });
+      });
+      it('should GET ALL Documents by the user defined from api/users/:id/documents', (done) => {
+        request
+          .get(`/api/users/${userId}/documents/`)
+          .set('x-access-token', token)
+          .expect('Content-Type', /json/)
+          .end((err, res) => {
+            expect(res.status).to.exist;
+            expect(res.body).to.exist;
+            expect(res.status).to.equal(200);
+            expect(res.body).to.be.a('object');
+            expect(res.body).to.have.property('document');
+            expect(Array.isArray(res.body.document)).to.be.true;
+            expect(res.body.document[1]).to.include.keys('_id', 'ownerId', 'content', 'title', 'modifiedAt', 'createdAt');
+            expect(res.body.document[1].ownerId).to.eql(res.body.document[0].ownerId);
+            done();
+          });
+      });
+      it('should GET ALL Documents by the role defined on them from api/documents?role=Admin', (done) => {
+        request
+          .get('/api/search/?role=Admin')
+          .set('x-access-token', token)
+          .expect('Content-Type', /json/)
+          .end((err, res) => {
+            expect(res.status).to.exist;
+            expect(res.body).to.exist;
+            expect(res.status).to.equal(200);
+            expect(res.body).to.be.a('object');
+            expect(res.body).to.have.property('document');
```

Comment:
**Code Quality - NitPick** 

Should return many documents so thus should be in plural. `documents` not documents